app/validators/routing/types.py

```diff
@@ -42,33 +42,67 @@
     "NoneType": TYPE_NULL,
 }
 
+METADATA_TYPE_TO_JSON_TYPE = {
+    "string": TYPE_STRING,
+    "date": TYPE_STRING,
+    "boolean": TYPE_BOOLEAN,
+}
+
+
+def resolve_answer_source_json_type(value_source, schema):
+    answer_id = value_source["identifier"]
+    answer_type = schema.answers_with_context[answer_id]["answer"]["type"]
+    return ANSWER_TYPE_TO_JSON_TYPE[answer_type]
+
+
+def resolve_calculated_summary_source_json_type(value_source, schema):
```

Comment:
based off my comment on identifier/source the arguments for this could then be 
`resolve_calculated_summary_source_json_type(block_id, schema):`